website/docs/d/google_compute_default_service_account.html.markdown

```diff
@@ -0,0 +1,32 @@
+---
+layout: "google"
+page_title: "Google: google_compute_default_service_account"
+sidebar_current: "docs-google-datasource-compute-default-service-account"
+description: |-
+  Retrieve default service account used by VMs running in this project
+---
+
+# google\_compute\_default\_service\_account
+
+Use this data source to retrieve default service account for this project
+
+## Example Usage
+
+```hcl
+data "google_compute_default_service_account" "default" { }
+
+output "default_account" {
+  value = "${google_compute_default_service_account.default.id}"
+} 
+```
+
+## Argument Reference
+
+There are no arguments available for this data source.
+
+
+## Attributes Reference
+
+The following attributes are exported:
+
+* `id` - Email address of the default service account used by VMs running in this project
```

Comment:
Add a field named `service_account` in the schema instead.

Reply:
I can do that, but the data source itself is named data "google_compute_default_service_account". Does it make sense to add a field "service_account"? "google_compute_default_service_account.service_account" seems a little repetitive to me

Reply:
@danawillow what are your thoughts on this?

To give you some context. The DefaultServiceAccount is retrieved by calling the compute Project GET API. Our `google_project` data source is getting the project details from the resource manager GET API. The response **does not** include the DefaultServiceAccount.

I see different options:
1. Create a `google_compute_default_service_account` resource (this PR). In that case, do we only set the id field or do we call the field a different name (e.g. service_account)?
2. Create a new resource to get all the compute GET API project details. In addition to the DefaultServiceAccount, we can also get the quotas, enabledFeatures and commoninstancemetadata. But how do we call this data source since we already have one for the resource manager GET API called `google_project`?
3. Make a call to the compute GET API in the project data source and include the data there.

Reply:
1. Interpolating based on ids isn't a good idea, since we've had many examples of changing ids for a resource over time. If we go with option one, then we should have it as a separate field even though it seems weird. In the `google_service_account` resource we have fields called `email` and `account_id`, so this one could match that.
2. If we did this, we could call it `google_compute_project` or `google_compute_project_info`. I don't love either of those names, but metadata is a totally separate thing we can't call it that.
3. The thing that makes me a bit nervous about this is that the info the compute projects resource returns is compute specific. For example, the quotas it returns are only the GCE quotas. I don't know if that will ever change.

Right now I'm leaning towards option 1 (and then just have separate data sources for each item in the project resources as needed), but could definitely be persuaded that option 2 is the way to go.

Reply:
Thanks for your input Dana :)

I am also leaning towards option 1. To be consistent with the `google_service_account` resource, we should add a new field called `email` and set the default service account email to this field.

In the future, if users ask for the default service account id, we can add a new field called `account_id` to this `google_compute_default_service_account` data source  and extract this value from the email.

Reply:
@rosbo @danawillow thanks for you input. I agree that having those to be consistent is a better approach.  commit with the requested changes is on its way.